flow/src/main/java/com/sequenceiq/flow/core/AbstractAction.java

```diff
@@ -74,24 +81,28 @@ public void execute(StateContext<S, E> context) {
             try {
                 Map<Object, Object> variables = context.getExtendedState().getVariables();
                 prepareExecution(payload, variables);
-                flowContext = createFlowContext(flowParameters, context, payload);
-                Object flowStartTime = variables.get(FLOW_START_TIME);
-                if (flowStartTime != null) {
-                    Object execTime = variables.get(FLOW_START_EXEC_TIME);
-                    long flowElapsed = (System.currentTimeMillis() - (long) flowStartTime) / MS_PER_SEC;
-                    long execElapsed = (System.currentTimeMillis() - (long) execTime) / MS_PER_SEC;
-                    String flowStateName = String.valueOf(variables.get(FLOW_STATE_NAME));
-                    long executionTime = Math.max(execElapsed, flowElapsed);
-                    LOGGER.debug("Stack: {}, flow state: {}, phase: {}, execution time {} sec", payload.getResourceId(),
-                            flowStateName, execElapsed > flowElapsed ? "doExec" : "service", executionTime);
-                    metricService.submit(FlowMetricType.FLOW_STEP, executionTime, Map.of("name", flowStateName.toLowerCase()));
+                String flowStateName = String.valueOf(variables.get(FLOW_STATE_NAME));
+                Span activeSpan = tracer.activeSpan();
+                String operationName = context.getEvent().name();
+                SpanContext spanContext = flowParameters.getSpanContext();
+                if (TracingUtil.isActiveSpanReusable(activeSpan, spanContext, operationName)) {
```

Comment:
could you refactor so both the if and else body is in a method?

Reply:
also some log would be nice to see what's happening around here

Reply:
its already hard to come up with a name for doExecute. Can you help me out with it?

Reply:
didnt find a way to do it. multiple exit points all the time

Reply:
you must find one, try it harder please!